flatpak/org.nanuc.Axolotl.yml

```diff
@@ -59,16 +59,14 @@ modules:
         # Cargo offline mode
         CARGO_NET_OFFLINE: "true"
     build-commands:
-      # Install the packages from the offline cache
-      - yarn --cwd=axolotl-web install
-
       # Build axolotl-web
-      - yarn --cwd=axolotl-web run build
+      - make build-dependencies-axolotl-web
+      - make build-axolotl-web
 
       # Build axolotl
-      - cargo --offline fetch --verbose
-      - cargo --offline build --features tauri --release --verbose
-      - install -Dm755 ./target/release/axolotl -t /app/bin/
+      - cargo fetch --verbose
+      - cargo build --features tauri --release --verbose
+      - install -Dm 755 target/release/axolotl -t ${FLATPAK_DEST}/bin/
```

Comment:
What I am trying to describe is to see the flatpak not as its own case, but to treat the build and install steps the same, either we are inside a Flatpak or outside. These steps are basically done for all packaging formats, and if we could re-use this logic we might easier both keep several packaging formats up to date, but also adopt new ones faster.

Reply:
here is another example of using the install-axolotl together with the Flatpak envs:

```
$ make install-axolotl CURRENT_DIR=target/release DESTDIR=${FLATPAK_DEST} INSTALL_PREFIX=/bin
install -D -m 755 target/release/axolotl /app/bin/axolotl/axolotl
```

Reply:
@olof-nord I agree that this would be desirable but given the fact that the `Makefile` was broken before I started my work on the Flatpak and given that also you didn't notice you broke the `Makefile` with the migration to `yarn` shows me that at least so far nobody is using the `Makefile` (neither humans nor other packaging systems).

Furthermore, I am not sure if "building and installing axolotl" and "running developer tools like downloading and installing dependencies" should go in the same file.

I suggest we move such developer tooling into a separate `tools/Makefile` such that the main one is only about building and installing axolotl. Ideally `make install` (+ variables) should be sufficient for all packaging systems. Another alternative would be to use `meson` for the main building+installing like many Rust+GNOME applications do and keep the `Makefile` only as developer scripts. What do you think?